pkg/clients/sls/index.go

```diff
@@ -104,8 +104,10 @@ func IsIndexNotFoundError(err error) bool {
 	if err == nil {
 		return false
 	}
-	if e, ok := errors.Cause(err).(*sdk.Error); ok && (e.Code == ErrCodeLogstoreIndexNotExist) {
-		return true
+	var slserr *sdk.Error
+	cause := errors.Cause(err)
+	if !errors.As(cause, &slserr) {
+		return false
 	}
-	return false
+	return slserr.Code == ErrCodeLogstoreIndexNotExist
```

Comment:
I think the above can be simplified to
```suggestion
	return errors.As(err, &slerr) && slserr.Code == ErrCodeLogstoreIndexNotExist
```

which should also be applicable in a few other places.